parla/task_runtime.py

```diff
@@ -327,20 +313,32 @@ def req(self):
     def req(self, new_req):
         self._req = new_req
 
+    def _wait_for_predecessor_events(self, env: TaskEnvironment):
+        if len(self._predecessor_events) > 0:
+            # Only wait events if any remaining dependent tasks exist.
+            # TODO(lhc): This does not support nested CPU tasks from GPU tasks.
+            #            When we notify successors, we don't know places on
+            #            which the successors will run since those would decide
+            #            at the next step, mapping step.
+            #            Therefore, the current runtime does not consider
+            #            successors' placements, but just creates events
+            #            on the devices where the current task is running.
+            #            For example, when CPU tasks get GPU events,
+            #            they will just skip and will not wait for them.
+            #            This is not technical limitation, just need refactoring
+            #            and adding more features.
+            #            But our target applications do not have this pattern
+            #            and will do it later.
+            env.wait_dependent_events(self._predecessor_events)
+            # Already waited all dependent events.
+            # Initialize the dependent event list.
+        self._predecessor_events = []
```

Comment:
Is the above function `_wait_for_predecessor_events` new, or did it replace something that already existed? What code should I compare with?